core/src/wsv.rs

```diff
@@ -367,6 +413,42 @@ impl<W: WorldTrait> WorldStateView<W> {
         &self.world.domains
     }
 
+    /// Get `Domain` and pass it to closure.
+    ///
+    /// # Errors
+    /// Fails if there is no domain
+    #[allow(clippy::panic_in_result_fn)]
+    pub fn map_domain<T>(
+        &self,
+        id: &<Domain as Identifiable>::Id,
+        f: impl FnOnce(&Domain) -> Result<T, Infallible>,
```

Comment:
In the interest of API broad-ness I would also consider making it a fallible function. Just because it doesn't return an `Err` today, doesn't mean that we won't need to use a fallible function in the future. It would also shorten the function to the point where it can be inlined.